tests/single_file/single_file_testing_utils.py

```diff
@@ -98,8 +99,8 @@ def test_single_file_components_local_files_only(self, pipe=None, single_file_pi
         pipe = pipe or self.pipeline_class.from_pretrained(self.repo_id, safety_checker=None)
 
         with tempfile.TemporaryDirectory() as tmpdir:
-            ckpt_filename = self.ckpt_path.split("/")[-1]
-            local_ckpt_path = download_single_file_checkpoint(self.repo_id, ckpt_filename, tmpdir)
+            repo_id, weight_name = _extract_repo_id_and_weights_name(self.ckpt_path)
```

Comment:
Feels like an overkill to me in using a method for just doing a splitting and indexing. 

Or is it more nuanced than that?

Reply:
Extracts the repo id and weight name from valid HF hosted repo urls 
https://github.com/huggingface/diffusers/blob/8ba90aa706a733f45d83508a5b221da3c59fe4cd/src/diffusers/loaders/single_file_utils.py#L293